cv_next/components/layout/navbar/navbar.tsx

```diff
@@ -16,51 +16,114 @@ const navLinks = [
 ];
 
 export default function Navbar() {
+  const [isMenuOpen, setIsMenuOpen] = useState(false);
+  const menuRef = useRef<HTMLDivElement>(null);
+  const buttonRef = useRef<HTMLButtonElement>(null);
+
+  const handleClick = () => {
+    setIsMenuOpen(!isMenuOpen);
+  };
+
+  const handleBlur = (e: React.FocusEvent) => {
+    if (
+      menuRef.current &&
+      !menuRef.current.contains(e.relatedTarget as Node) &&
+      buttonRef.current &&
+      !buttonRef.current.contains(e.relatedTarget as Node)
+    ) {
+      setIsMenuOpen(false);
+    }
+  };
+
   return (
     <>
       <header className="relative top-0 h-[72px] w-full select-none">
         <div className="fixed top-0 -z-50 h-[36px] w-full translate-y-2/4 bg-white" />
-        <div className="fixed top-0 z-40 h-[72px] w-full bg-black bg-opacity-75 backdrop-blur-2xl">
-          {" "}
-        </div>
+        <div className="fixed top-0 z-40 h-[72px] w-full bg-black bg-opacity-75 backdrop-blur-2xl" />
```

Comment:
Beware these pixel values

Reply:
Tested these on most of the emulators that the dev tools provide, looked good
If there's any way to do it better without the pixel values lmk

Reply:
Found a better way :)